src/custom_nodes/model_zoo_intel_object_detection/model_zoo_intel_object_detection.cpp

```diff
@@ -58,12 +95,16 @@ bool copy_images_into_output(struct CustomNodeTensor* output, const std::vector<
             std::memcpy(buffer + (i * channels * targetImageWidth * targetImageHeight), image.data, byteSize / outputBatch);
         }
     }
-
     output->data = reinterpret_cast<uint8_t*>(buffer);
     output->dataBytes = byteSize;
+
+    u_int64_t* dims = nullptr;
+    if (!get_preallocated_buffer<u_int64_t>(internalManager, &dims, OUTPUT_IMAGES_DIMS_NAME, 5 * sizeof(uint64_t))) {
+        release(buffer, internalManager);
+        return false;
```

Comment:
We need to release `float* buffer = nullptr;` here in case of failure as well, otherwise there is memory leak

Reply:
As if for now if dims allocation fails then we are calling release on the exact buffer you pointed in line 103. Do i miss something?